Automattic/wp-calypso
# Tweak post dates and other styles in `PostTypeList` and `PostItem`

This PR is part of a larger epic and behind the `posts/post-type-list` feature flag.  See p8F9tW-hL-p2.

It changes the `PostTypeList` display in the following ways:

- Uses full dates like "August 30, 2017 4:46 PM" for posts older than 1 week
- Removes the clock icon from the post date display

This PR renames `PostRelativeTime` to `PostTime` since the old name is no longer accurate, and makes the needed changes inside that component.

This changes the display of the `PostItem` component, which is also used in production to show the drafts list of the current (non-feature-flagged) `/posts` view.  Here is how this section of the screen changes as a result - I think the changes are acceptable, and preferable to maintaining branching logic inside the `PostTime` component:

| Before | After |
|--|--|
| ![2017-08-30t17 19 39-0500](https://user-images.githubusercontent.com/227022/29897619-70b74eac-8da7-11e7-96fa-c93b2e43c1d4.png) | ![2017-08-30t17 21 35-0500](https://user-images.githubusercontent.com/227022/29897676-b0720672-8da7-11e7-9236-7020addfb0ce.png) |

### `PostTypeList` before screenshot

![2017-08-30t17 16 37-0500](https://user-images.githubusercontent.com/227022/29897557-06511b56-8da7-11e7-9da4-89cf2268e53f.png)

### `PostTypeList` after screenshot

![2017-08-30t17 07 19-0500](https://user-images.githubusercontent.com/227022/29897438-77455634-8da6-11e7-8ea9-4594720caf16.png)

### To test

- Check out this branch
- Start Calypso with `ENABLE_FEATURES=posts/post-type-list npm start`
- Navigate to Blog Posts
- Verify that for recent post dates, the date is shown in relative terms like "2 days ago" (as before)
- Verify that for post dates 1 week ago or older, the date is shown in absolute terms like "August 30, 2017 4:46 PM"
- Verify that post dates no longer show a small clock icon on their left side.
